src/datascience-ui/data-explorer/mainPanel.tsx

```diff
@@ -233,6 +264,8 @@ export class MainPanel extends React.Component<IMainPanelProps, IMainPanelState>
                 const indexColumn = variable.indexColumn ? variable.indexColumn : 'index';
                 const originalVariableType = this.state.originalVariableType ?? variable.type;
                 const originalVariableShape = this.state.originalVariableShape ?? variable.shape;
+                const variableName = this.state.variableName ?? variable.name;
+                const fileName = this.state.fileName ?? variable.fileName;
```

Comment:
Yeah, its weird for a variable to have a file name. I'd prefer this to be sent as a separate entity.

Reply:
Is it really that weird though? The variable must've been declared somewhere. I think it should definitely be renamed but it doesn't seem that weird to me to have `backingFileName` or similar as an attribute associated with the variable info.

Reply:
It is weird, if you look at the debugger prototcol or others, variables are not attached to a file name. We have a file name that can contains a context & that context then contains a list of variables & that list contains the individual variable.
I.e. the mapping is at a much higher level & this seem wrong to me.
But don't want to block this PR for that.

I.e. the request object (sending request to kernel for variables) should already know the file name that is active.
Also if we're displaying variables for Notebooks, would we display the file name as some internal `python file` or the name of the notebook? Today run by line works in webview, if we use run by line & display the variables, what do you see in the file name (breadcrumb)?

Reply:
Not worth blocking PR

Reply:
But i'd test this with run by line & the others, not sure the name of the file would be as expected/required. I.e. in notebooks I'd always expect to see the name of the notebook if anything.

Reply:
The request we send to the kernel doesn't ask for the filename. The filename property is only included on the variable at the point when we send the variable info to the UI for display.

For webview notebooks the file name is the name of the notebook, e.g.
![image](https://user-images.githubusercontent.com/30305945/108886866-55e15980-75be-11eb-801b-f6bd5dade18b.png). This is taken from `notebook.identity.path`.

For RBL I initially went with `debugLocation` because `notebook.identity` is not available when debugging a Python file, but this doesn't work for RBL, the filename would've been '<ipython input>'. Changed debuggerVariables.ts to use the notebook.identity.path when it's available (which it is for Run By Line), and use the debugLocation when it's not (i.e. debugging a Python file). Good catch :) 

![image](https://user-images.githubusercontent.com/30305945/108888120-ebc9b400-75bf-11eb-9745-effa6102c950.png)